ggml.c

```diff
@@ -2542,6 +2542,12 @@ static inline bool ggml_is_padded_1d(const struct ggml_tensor * tensor) {
         tensor->nb[3] == tensor->nb[2]*tensor->ne[2];
 }
 
+static inline bool ggml_is_empty(const struct ggml_tensor * tensor) {
+    // the stride of the last dimension is zero if any of the previous dimensions has no elements
+    // but to be sure, the number of elements in the last dimension must also be checked
+    return tensor->nb[GGML_MAX_DIMS - 1] == 0 || tensor->ne[GGML_MAX_DIMS - 1] == 0;
```

Comment:
I think it is good to add support for empty tensors, it will allow writing code with fewer special cases, but it will require updating all the backends to skip empty ops. To facilitate that, this function should be exposed in `ggml.h`.

This implementation will not work reliably with views, do something like this instead:

```suggestion
    for (int i = 0; i < GGML_MAX_DIMS; ++i) {
        if (tensor->ne[i] == 0) {
            return true;
        }
    }
    return false;
```